### DISCUSSION TITLE
Ajax Validation - Not sure if this is a

### DISCUSSION THREAD

**Discussion by tillmannschiffler** [yiisoft/yii2#20109] [General]
I have a problem that I'm not sure if it's a bug or not. I created a form in Yii2 with quite a few input options, some of which have cascading validation rules. I'm using Ajax validation, which correctly returns a response with my messages. However, only the active changed field with the error message is updated in the browser. All other fields, even though they have updated or new error messages, are not being updated. I have a feeling that in the 'activeForm.js,' all other fields are being skipped

**Comment by bizley** [comment]
Have you also got `enableClientValidation` in the form set to `true`? With that client side validation is performed first (using the JS) and if it is successful AJAX request is send, but this one is not triggering JS.

**Comment by tillmannschiffler** [comment]
Tested both with true / false. I even build a small demo to test various setups. As i mentioned before i got a json response from my controller with error messages for all fields up only the blured field got updated. I looked into the activeForm.js and found this lines:

```
            $.each(data.attributes, function () {
                if (!this.cancelled && (this.status === 2 || this.status === 3)) {
                    updateInput($form, this, messages);
                }
            });
```
It seems that only my blured field got status 3. I just the the if statement to ignore the status

```

            $.each(data.attributes, function () {
                if (!this.cancelled) {
                    updateInput($form, this, messages);
                }
            });
```

and all my fields got updated with the new error message
